src/renderer/api/kube-api.ts

```diff
@@ -174,4 +175,38 @@ export class KubeApi<T extends KubeObject = any> {
   }
 }
 
+export class VersionedKubeApi<T extends KubeObject = any> extends KubeApi<T> {
+  
+  /** A map of the upper bounds of a version and the previous API version */
+  versionBelow: Record<string, string>;
+
+  constructor(protected options: IKubeApiOptions<T> & { versionBelow: Record<string, string>} ) {
+    super(options);
+
+    this.versionBelow = options.versionBelow;
+  }
+
+  getUrl({ name = "", namespace = "" } = {}, query?: Partial<IKubeApiQueryParams>) {
+    const { apiPrefix, apiGroup, apiVersion, apiResource } = this;
+
+    const clusterVersion = getHostedCluster().version;
+
+    let version = apiVersion;
+
+    for (const versionCheck of Object.keys(this.versionBelow)) {
+      if (clusterVersion < versionCheck) version = this.versionBelow[versionCheck];
```

Comment:
Maybe better to use `semver` to check versions? (already added in `package.json`)

Reply:
Thanks for the tip - will give this a go